pkg/terminal/command.go

```diff
@@ -1462,13 +1465,30 @@ func printcontext(t *Term, state *api.DebuggerState) error {
 		fmt.Println("No current thread available")
 		return nil
 	}
-	if len(state.CurrentThread.File) == 0 {
+
+	var th *api.Thread
+	if state.SelectedGoroutine == nil {
+		th = state.CurrentThread
+	} else {
+		for i := range state.Threads {
+			if state.Threads[i].ID == state.SelectedGoroutine.ThreadID {
+				th = state.Threads[i]
+				break
+			}
+		}
+		if th == nil {
+			printcontextLocation(state.SelectedGoroutine.CurrentLoc)
+			return nil
+		}
+	}
+
+	if len(th.File) == 0 {
```

Comment:
nit: I know this is carry-over but let's be consistent and check against empty string instead of length.